### PR TITLE
✨ Introduces archive page for FAQs

### LINKED ISSUE #3: add initial archive page for FAQ
- [x] Setup initial layout for the FAQ page
- [x] prepare an archive page to list all FAQ content

### PR DESCRIPTION
**Pull request by ashiishme**
Adds archive page for FAQs content.

resolves: #3

### CODE REVIEW

File: src/utils/faqGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+const faqGenerator = () => {
+  const getFAQs = () => {
+    const files = fs.readdirSync(path.join(ROOT_DIR, POST_DIR));
+
+    if (!files.length) {
+      return null;
+    }
+
+    const faqs = files.map((file) => {
```
Comment (aabishkaryal) [inline src/utils/faqGenerator.ts]:
```ts
   const mdxFiles = files.filter(
      (file) => path.extname(file) === MDX_EXTENSION
    );
    const faqs = mdxFiles.map((file) => {
      const source = fs.readFileSync(path.join(ROOT_DIR, POST_DIR, file), {
        encoding: "utf-8",
      });

      const { data } = matter(source);

      const faqList = data as FAQProps;
      return {
        ...faqList,
        link: file.replace(MDX_EXTENSION, ""),
      };
    });
```
This provides better typesupport and is more clear.

File: src/components/FaqList.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import faqGenerator from "../utils/faqGenerator";
+import formatDate from "../utils/formatDate";
+
+const FaqList = () => {
+  const { getFAQs } = faqGenerator();
+  const faqs = getFAQs();
+  return (
+    <div className="mt-6 flex max-w-4xl flex-wrap items-center justify-around sm:w-full">
+      <ul>
+        {faqs?.map((faq) => {
```
Comment (aabishkaryal) [inline src/components/FaqList.tsx]:
``` ts
 {faqs?.map(({ title, date, link }) => { ..... }
 ```
This can be replaced as such after the changes

Comment (ashiishme) [reply src/components/FaqList.tsx]: Ts expects that `faq` can be undefined so direct destructing like you suggested wont work without null assertion.

Comment (aabishkaryal) [reply src/components/FaqList.tsx]: It can work after the changes from the previous review. https://github.com/askbuddie/faq/pull/5#discussion_r1035660195

**Review by aabishkaryal — DISMISSED** [review]

File: src/utils/faqGenerator.ts (hunk 1 above)
Comment (ashiishme) [reply src/utils/faqGenerator.ts]: Totally, super vai.

File: src/components/FaqList.tsx (hunk 2 above)
Comment (ashiishme) [reply src/components/FaqList.tsx]: Now I realize that too haha

**Review by aabishkaryal — APPROVED** [review]

**Review by aabishkaryal — APPROVED** [review]

**Pull request merged**